src/util-inl.h

```diff
@@ -229,30 +229,74 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
   return true;
 }
 
+template <typename T>
+T MultiplyWithOverflowCheck(T a, T b) {
+  T ret = a * b;
+  if (a != 0)
+    CHECK_EQ(b, ret / a);
```

Comment:
Looking at this again, I think this is dead code.  The branch is only taken when `a == 0` but `a` is always `sizeof(T)` and therefore never zero.